.gitlab-ci.yml

```diff
@@ -212,6 +212,22 @@ run_security_scan_test:
     - SNYK_TOKEN=$(aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.snyk_token --with-decryption --query "Parameter.Value" --out text)
       snyk monitor --project-name=datadog-agent-gopkg.lock --file=Gopkg.lock
 
+# check consistency of Gopkg.lock
+check_dep_gopkglock_consistency:
+  stage: source_test
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/deb_x64:latest
+  tags: [ "runner:main", "size:large" ]
+  before_script:
+    - cd $SRC_PATH
+    - pip install --upgrade --ignore-installed pip setuptools
+    - pip install -r requirements.txt
+  script:
+    - inv -e deps --verbose --dep-no-vendor
+    - git_diff="$(git diff-files -p Gopkg.lock)"
+    # if there's a change on Gopkg.lock, that means it's not in sync and should be updated: print error message and fail.
+    - test -z "$git_diff" || printf "Gopkg.lock is not in sync with Gopkg.toml and project imports. Please run inv deps and commit the change on Gopkg.lock. Diff:\n$git_diff\n"
```

Comment:
I think:
```
git_diff="$(git diff-files -p Gopkg.lock | wc - l)"
test "0" -eq "$git_diff" || printf "Gopkg.lock is not in sync with Gopkg.toml and project imports. Please run inv deps and commit the change on Gopkg.lock. Diff:\n$git_diff\n"
```
would have the desired effect.
`test -z` won't work with `wc -l` because `wc -l` returns a non-empty string, and `test -z` only matches an empty or unset variable

Reply:
Or even cleaner:
```
    - git_diff="$(git diff-files -p Gopkg.lock)"
    - git_diff_number="$(echo $git_diff | wc -l)"
    - test "0" -eq "$git_diff_number" || printf "Gopkg.lock is not in sync with Gopkg.toml and project imports. Please run inv deps and commit the change on Gopkg.lock. Diff: \n$git_diff\n"
```
should avoid using git diff-files twice

Reply:
I'm not using `wc -l` at all at the moment, just relying on the fact that `git diff-files -p Gopkg.lock` has an empty output unless there's indeed a change on that file.

But yeah I guess using `wc -l` is more explicit, I'll update accordingly

Reply:
Both solutions are fine actually 😄, pick the one you prefer. Yours has the advantage of not calling `diff-files` twice, and even if it's not as explicit, adding a small comment should be enough to understand the logic.